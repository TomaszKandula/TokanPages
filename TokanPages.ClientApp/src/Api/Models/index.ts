export type { AddArticleDto } from "./Articles/addArticleDto";
export type { UpdateArticleContentDto } from "./Articles/updateArticleContentDto";
export type { UpdateArticleCountDto } from "./Articles/updateArticleCountDto";
export type { UpdateArticleLikesDto } from "./Articles/updateArticleLikesDto";
export type { UpdateArticleVisibilityDto } from "./Articles/updateArticleVisibilityDto";
export type { RemoveArticleDto } from "./Articles/removeArticleDto";

export type { SendMessageDto } from "./Mailer/sendMessageDto";
export type { SendNewsletterDto } from "./Mailer/sendNewsletterDto";
export type { SubscriberInfoDto } from "./Mailer/subscriberInfoDto";
export type { VerifyEmailAddressDto } from "./Mailer/verifyEmailAddressDto";

export type { AddSubscriberDto } from "./Subscribers/addSubscriberDto";
export type { UpdateSubscriberDto } from "./Subscribers/updateSubscriberDto";
export type { RemoveSubscriberDto } from "./Subscribers/removeSubscriberDto";

export type { UserDataDto } from "./Users/userDataDto";
export type { ActivateUserDto } from "./Users/activateUserDto";
export type { AddUserDto } from "./Users/addUserDto";
export type { UpdateUserDto } from "./Users/updateUserDto";
export type { RemoveUserDto } from "./Users/removeUserDto";
export type { AuthenticateUserDto } from "./Users/authenticateUserDto";
export type { AuthenticateUserResultDto } from "./Users/authenticateUserResultDto";
export type { ReAuthenticateUserDto } from "./Users/reAuthenticateUserDto";
export type { ReAuthenticateUserResultDto } from "./Users/reAuthenticateUserResultDto";
export type { RevokeUserRefreshTokenDto } from "./Users/revokeUserRefreshTokenDto";
export type { ResetUserPasswordDto } from "./Users/resetUserPasswordDto";
export type { UpdateUserPasswordDto } from "./Users/updateUserPasswordDto";
export type { UserPermissionDto } from "./Users/userPermissionDto";
export type { UserRoleDto } from "./Users/userRoleDto";
export type { UploadUserMediaDto } from "./Users/uploadUserMediaDto";
export type { UploadUserMediaResultDto } from "./Users/uploadUserMediaResultDto";

export type { ErrorDto } from "./Error/errorDto";
export type { ValidationErrorsDto } from "./Error/validationErrorsDto";

export type { DocumentContentDto } from "./Components/documentContentDto";

export type { NavigationContentDto } from "./Components/navigationContentDto";
export type { HeaderContentDto } from "./Components/headerContentDto";
export type { FooterContentDto } from "./Components/footerContentDto";
export type { ClientsContentDto } from "./Components/clientsContentDto";
export type { ArticleFeaturesContentDto } from "./Components/articleFeaturesContentDto";
export type { ContactFormContentDto } from "./Components/contactFormContentDto";
export type { CookiesPromptContentDto } from "./Components/cookiesPromptContentDto";
export type { FeaturedContentDto } from "./Components/featuredContentDto";
export type { FeaturesContentDto } from "./Components/featuresContentDto";
export type { NewsletterContentDto } from "./Components/newsletterContentDto";
export type { ResetPasswordContentDto } from "./Components/resetPasswordContentDto";
export type { UpdatePasswordContentDto } from "./Components/updatePasswordContentDto";
export type { UserSigninContentDto } from "./Components/userSigninContentDto";
export type { UserSignoutContentDto } from "./Components/userSignoutContentDto";
export type { UserSignupContentDto } from "./Components/userSignupContentDto";
export type { TestimonialsContentDto } from "./Components/testimonialsContentDto";
export type { UnsubscribeContentDto } from "./Components/unsubscribeContentDto";
export type { ActivateAccountContentDto } from "./Components/activateAccountContentDto";
export type { UpdateSubscriberContentDto } from "./Components/updateSubscriberContentDto";
export type { WrongPagePromptContentDto } from "./Components/wrongPagePromptContentDto";
export type { 
    AccountContentDto, 
    SectionAccessDenied,
    SectionAccountInformation,
    SectionAccountPassword,
    SectionAccountRemoval
} from "./Components/accountContentDto";

export type { IconDto } from "./Components/Common/iconDto";
export type { LinkDto } from "./Components/Common/linkDto";
export type { RowItemDto } from "./Components/Common/rowItemDto";
export type { TextItemDto } from "./Components/Common/textItemDto";
export type { ContentDto } from "./Components/Common/contentDto";

export type { GetContentManifestDto } from "./Content/getContentManifestDto";
export type { LanguageItemDto } from "./Content/Items/languageItemDto";
export type { JWT } from "./Jwt";
