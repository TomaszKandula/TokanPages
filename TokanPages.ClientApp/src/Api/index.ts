export {
    IsSuccessStatusCode,
    GetBaseHeaders,
    GetProcessedBody,
    GetProcessedHeaders,
    GetProcessedResponse,
} from "./Utilities";

export type { ExecuteApiActionResultProps, ExecuteStoreActionProps, ExecuteApiActionProps } from "./Utilities/Types";

export {
    API_BASE_URI,
    GET_ARTICLES,
    GET_ARTICLE,
    GET_ARTICLE_BY_TITLE,
    ADD_ARTICLE,
    UPDATE_ARTICLE_CONTENT,
    UPDATE_ARTICLE_COUNT,
    UPDATE_ARTICLE_LIKES,
    UPDATE_ARTICLE_VISIBILITY,
    RETRIEVE_ARTICLE_INFO,
    REMOVE_ARTICLE,
    LOG_MESSAGE,
    GET_USERS,
    GET_USER,
    GET_USER_IMAGE,
    ADD_USER,
    ACTIVATE_USER,
    UPDATE_USER,
    REMOVE_USER,
    AUTHENTICATE,
    REAUTHENTICATE,
    VERIFY_USER_EMAIL,
    REVOKE_REFRESH_TOKEN,
    REVOKE_USER_TOKEN,
    RESET_USER_PASSWORD,
    UPDATE_USER_PASSWORD,
    UPLOAD_USER_IMAGE,
    GET_NEWSLETTERS,
    GET_NEWSLETTER,
    ADD_NEWSLETTER,
    UPDATE_NEWSLETTER,
    REMOVE_NEWSLETTER,
    SEND_MESSAGE,
    SEND_NEWSLETTER,
    NOTIFY_WEB_URL,
    NOTIFICATION_STATUS,
    GET_CONTENT_MANIFEST,
    REQUEST_PAGE_DATA,
    GET_ARTICLE_MAIN_IMAGE_URL,
    GET_IMAGES_URL,
    GET_DOCUMENTS_URL,
    ARTICLE_PATH,
    GET_USER_VIDEO,
    GET_VIDEO_ASSET,
    GET_USER_NOTES,
    GET_USER_NOTE,
    ADD_USER_NOTE,
    UPDATE_USER_NOTE,
    REMOVE_USER_NOTE,
} from "./Paths";
