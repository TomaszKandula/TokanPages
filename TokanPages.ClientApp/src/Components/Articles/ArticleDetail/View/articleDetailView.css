.article-details-avatar {
    background-color: #9e9e9e;
}

.article-details-text-block {
    display: flex;
    justify-content: left;
    flex-direction: row;
}

.article-details-text-padding-left {
    padding-left: 5px;
}

.article-details-flag-image {
    height: 18px;
    width: 18px;
    display: flex;
    align-self: center;
}

.article-details-alias-name {
    padding-top: 10px;
    font-weight: bold !important;
}

.article-details-thumbs-medium {
    color: #9e9e9e;
    cursor: pointer;
    width: 24px;
    height: 24px;
}

.article-details-likes-tip {
    font-size: 1.5em;
}

.article-details-popover {
    pointer-events: none;
}

.article-details-user {
    margin-top: 16px;
    margin-bottom: 16px;
    margin-left: 24px;
    margin-right: 24px;
}

.article-details-text-500 {
    font-weight: 500;
}
