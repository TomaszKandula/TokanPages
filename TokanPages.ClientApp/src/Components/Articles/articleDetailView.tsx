import * as React from "react";
import Container from "@material-ui/core/Container";
import Box from "@material-ui/core/Box";
import { Divider, Grid, IconButton, Popover, Tooltip, Typography } from "@material-ui/core";
import { ArrowBack } from "@material-ui/icons";
import ThumbUpIcon from "@material-ui/icons/ThumbUp";
import Emoji from "react-emoji-render";
import articleDetailStyle from "./Styles/articleDetailStyle";
import { FormatDateTime } from "../../Shared/helpers";

interface IBinding 
{
    bind: IProperties;
}

interface IProperties
{
    backButtonHandler: any;
    articleReadCount: number;
    openPopoverHandler: any;
    closePopoverHandler: any;
    renderSmallAvatar: JSX.Element;
    renderLargeAvatar: JSX.Element;
    authorAliasName: string;
    popoverOpen: boolean;
    popoverElement: HTMLElement | null;
    authorFirstName: string;
    authorLastName: string;
    authorRegistered: string;
    articleReadTime: string;
    articleCreatedAt: string;
    articleUpdatedAt: string;
    articleContent: JSX.Element;
    renderLikesLeft: string;
    thumbsHandler: any;
    totalLikes: number;
    renderAuthorName: string;
    authorShortBio: string;
}

const ArticleDetailView = (props: IBinding): JSX.Element =>
{
    const classes = articleDetailStyle();
    return (
        <section>
            <Container className={classes.container}>
                <Box py={12}>
                    <div data-aos="fade-down">
                        <Grid container spacing={3}>
                            <Grid item xs={6}>
                                <IconButton onClick={props.bind?.backButtonHandler}>
                                    <ArrowBack  /> 
                                </IconButton>
                            </Grid>
                            <Grid item xs={6}>
                                <Typography className={classes.readCount} component="p" variant="subtitle1" align="right">
                                    Read: {props.bind?.articleReadCount}
                                </Typography>
                            </Grid>
                        </Grid>
                        <Divider className={classes.dividerTop} />
                        <Grid container spacing={2}>
                            <Grid item>
                                <Box onMouseEnter={props.bind?.openPopoverHandler} onMouseLeave={props.bind?.closePopoverHandler}>
                                    {props.bind?.renderSmallAvatar}
                                </Box>
                            </Grid>
                            <Grid item xs zeroMinWidth>
                                <Typography className={classes.aliasName} component="div" variant="subtitle1" align="left">
                                    <Box fontWeight="fontWeightBold">
                                        {props.bind?.authorAliasName}
                                    </Box>
                                </Typography>
                                <Popover
                                    id="mouse-over-popover"
                                    className={classes.popover}
                                    open={props.bind?.popoverOpen}
                                    anchorEl={props.bind?.popoverElement}
                                    anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
                                    transformOrigin={{ vertical: "top", horizontal: "left" }}
                                    onClose={props.bind?.closePopoverHandler}
                                    disableRestoreFocus
                                >
                                    <Box mt={2} mb={2} ml={3} mr={3} >
                                        <Typography component="p" variant="subtitle2" color="textSecondary">
                                            First name: {props.bind?.authorFirstName}
                                        </Typography>
                                        <Typography component="p" variant="subtitle2" color="textSecondary">
                                            Last name: {props.bind?.authorLastName}
                                        </Typography>
                                        <Typography component="p" variant="subtitle2" color="textSecondary">
                                            Registered at: {props.bind?.authorRegistered}
                                        </Typography>
                                    </Box>
                                </Popover>
                            </Grid>
                        </Grid>
                        <Box mt={1} mb={5}>
                            <Typography component="p" variant="subtitle1">
                                Read time: {props.bind?.articleReadTime} min.
                            </Typography>
                            <Typography component="p" variant="subtitle1">
                                Published at: {FormatDateTime(props.bind?.articleCreatedAt, true)}
                            </Typography>
                            <Typography component="p" variant="subtitle2" color="textSecondary">
                                Updated at: {FormatDateTime(props.bind?.articleUpdatedAt, true)}
                            </Typography>
                        </Box>
                    </div>
                    <div data-aos="fade-up">
                        {props.bind?.articleContent}
                    </div>
                    <Box mt={5}>
                        <Grid container spacing={2}>
                            <Grid item>
                                <Tooltip title=
                                    {<span className={classes.likesTip}>
                                        {<Emoji text={props.bind?.renderLikesLeft}/>}
                                    </span>} arrow>
                                    <ThumbUpIcon className={classes.thumbsMedium} onClick={props.bind?.thumbsHandler} />
                                </Tooltip>
                            </Grid>
                            <Grid item xs zeroMinWidth>
                                <Typography component="p" variant="subtitle1">
                                    {props.bind?.totalLikes}
                                </Typography>
                            </Grid>
                        </Grid>
                    </Box>
                    <Divider className={classes.dividerBottom} />
                    <Grid container spacing={2}>
                        <Grid item>
                            {props.bind?.renderLargeAvatar}
                        </Grid>
                        <Grid item xs zeroMinWidth>
                            <Typography className={classes.aliasName} component="span" variant="h6" align="left" color="textSecondary">
                                Written by
                            </Typography>
                            <Box fontWeight="fontWeightBold">
                                <Typography className={classes.aliasName} component="span" variant="h6" align="left">
                                    {props.bind?.renderAuthorName}
                                </Typography>
                            </Box>
                            <Typography className={classes.aliasName} component="span" variant="subtitle1" align="left" color="textSecondary">
                                About the author: {props.bind?.authorShortBio}
                            </Typography>
                        </Grid>
                    </Grid>
                </Box>
            </Container>
        </section>
    );
}

export default ArticleDetailView;
