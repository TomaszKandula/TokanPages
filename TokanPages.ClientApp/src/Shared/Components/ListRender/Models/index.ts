export type { Fields } from "./Fields";
export type { Item } from "./Item";
export type { Subitem } from "./Subitem";
