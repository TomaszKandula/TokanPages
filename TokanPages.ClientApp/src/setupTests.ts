// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
