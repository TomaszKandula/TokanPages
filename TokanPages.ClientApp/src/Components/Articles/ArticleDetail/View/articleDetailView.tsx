import * as React from "react";
import Container from "@material-ui/core/Container";
import Box from "@material-ui/core/Box";
import { Divider, Grid, IconButton, Popover, Tooltip, Typography } from "@material-ui/core";
import { ArrowBack } from "@material-ui/icons";
import ThumbUpIcon from "@material-ui/icons/ThumbUp";
import Emoji from "react-emoji-render";
import { GET_FLAG_URL } from "../../../../Api/Request";
import { RenderImage } from "../../../../Shared/Components";
import { GetDateTime } from "../../../../Shared/Services/Formatters";
import { ReactMouseEvent } from "../../../../Shared/types";
import { ArticleDetailStyle } from "./articleDetailStyle";

interface Properties {
    backButtonHandler: () => void;
    articleReadCount: number;
    openPopoverHandler: (event: ReactMouseEvent) => void;
    closePopoverHandler: () => void;
    renderSmallAvatar: JSX.Element;
    renderLargeAvatar: JSX.Element;
    authorAliasName: string;
    popoverOpen: boolean;
    popoverElement: HTMLElement | null;
    authorFirstName: string;
    authorLastName: string;
    authorRegistered: string;
    articleReadTime: string;
    articleCreatedAt: string;
    articleUpdatedAt: string;
    articleContent: JSX.Element;
    renderLikesLeft: string;
    thumbsHandler: any;
    totalLikes: number;
    renderAuthorName: string;
    authorShortBio: string;
    flagImage: string;
}

export const ArticleDetailView = (props: Properties): JSX.Element => {
    const classes = ArticleDetailStyle();
    return (
        <section className={classes.section}>
            <Container className={classes.container}>
                <Box py={12}>
                    <div data-aos="fade-down">
                        <Grid container spacing={3}>
                            <Grid item xs={6}>
                                <IconButton onClick={props.backButtonHandler}>
                                    <ArrowBack />
                                </IconButton>
                            </Grid>
                            <Grid item xs={6}>
                                <Typography
                                    className={classes.readCount}
                                    component="p"
                                    variant="subtitle1"
                                    align="right"
                                >
                                    Read: {props.articleReadCount}
                                </Typography>
                            </Grid>
                        </Grid>
                        <Divider className={classes.dividerTop} />
                        <Grid container spacing={2}>
                            <Grid item>
                                <Box onMouseEnter={props.openPopoverHandler} onMouseLeave={props.closePopoverHandler}>
                                    {props.renderSmallAvatar}
                                </Box>
                            </Grid>
                            <Grid item xs zeroMinWidth>
                                <Typography
                                    className={classes.aliasName}
                                    component="div"
                                    variant="subtitle1"
                                    align="left"
                                >
                                    <Box fontWeight="fontWeightBold">{props.authorAliasName}</Box>
                                </Typography>
                                <Popover
                                    id="mouse-over-popover"
                                    className={classes.popover}
                                    open={props.popoverOpen}
                                    anchorEl={props.popoverElement}
                                    anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
                                    transformOrigin={{ vertical: "top", horizontal: "left" }}
                                    onClose={props.closePopoverHandler}
                                    disableRestoreFocus
                                >
                                    <Box mt={2} mb={2} ml={3} mr={3}>
                                        <Typography component="p" variant="subtitle2" color="textSecondary">
                                            First name: {props.authorFirstName}
                                        </Typography>
                                        <Typography component="p" variant="subtitle2" color="textSecondary">
                                            Last name: {props.authorLastName}
                                        </Typography>
                                        <Typography component="p" variant="subtitle2" color="textSecondary">
                                            Registered at: {props.authorRegistered}
                                        </Typography>
                                    </Box>
                                </Popover>
                            </Grid>
                        </Grid>
                        <Box mt={1} mb={5}>
                            <Typography component="p" variant="subtitle1">
                                Language: {RenderImage(GET_FLAG_URL, props.flagImage, classes.flag_image)}
                            </Typography>
                            <Typography component="p" variant="subtitle1">
                                Read time: {props.articleReadTime} min.
                            </Typography>
                            <Typography component="p" variant="subtitle1">
                                Published at: {GetDateTime({ value: props.articleCreatedAt, hasTimeVisible: true })}
                            </Typography>
                            <Typography component="p" variant="subtitle2" color="textSecondary">
                                Updated at: {GetDateTime({ value: props.articleUpdatedAt, hasTimeVisible: true })}
                            </Typography>
                        </Box>
                    </div>
                    <div data-aos="fade-up">{props.articleContent}</div>
                    <Box mt={5}>
                        <Grid container spacing={2}>
                            <Grid item>
                                <Tooltip
                                    title={
                                        <span className={classes.likesTip}>
                                            {<Emoji text={props.renderLikesLeft} />}
                                        </span>
                                    }
                                    arrow
                                >
                                    <ThumbUpIcon className={classes.thumbsMedium} onClick={props.thumbsHandler} />
                                </Tooltip>
                            </Grid>
                            <Grid item xs zeroMinWidth>
                                <Typography component="p" variant="subtitle1">
                                    {props.totalLikes}
                                </Typography>
                            </Grid>
                        </Grid>
                    </Box>
                    <Divider className={classes.dividerBottom} />
                    <Grid container spacing={2}>
                        <Grid item>{props.renderLargeAvatar}</Grid>
                        <Grid item xs zeroMinWidth>
                            <Typography
                                className={classes.aliasName}
                                component="span"
                                variant="h6"
                                align="left"
                                color="textSecondary"
                            >
                                Written by
                            </Typography>
                            <Box fontWeight="fontWeightBold">
                                <Typography className={classes.aliasName} component="span" variant="h6" align="left">
                                    {props.renderAuthorName}
                                </Typography>
                            </Box>
                            <Typography
                                className={classes.aliasName}
                                component="span"
                                variant="subtitle1"
                                align="left"
                                color="textSecondary"
                            >
                                About the author: {props.authorShortBio}
                            </Typography>
                        </Grid>
                    </Grid>
                </Box>
            </Container>
        </section>
    );
};
