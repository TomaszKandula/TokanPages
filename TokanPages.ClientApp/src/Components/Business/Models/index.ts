import { ServiceItemDto, TechItemsDto } from "../../../Api/Models";
import { ReactChangeEvent, ReactMouseEvent } from "../../../Shared/types";

export interface MessageFormProps {
    company: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    description: string;
    techStack: string[];
    services: string[];
}

export interface TechStackListProps {
    isDisabled: boolean;
    list: TechItemsDto[];
    handler: (event: ReactChangeEvent, isChecked: boolean) => void;
}

export interface ServiceItemCardProps {
    isDisabled: boolean;
    key: React.Key | null | undefined;
    value: ServiceItemDto;
    handler: (event: ReactMouseEvent) => void;
    services: string[];
}

export interface BusinessFormProps {
    hasCaption?: boolean;
    hasIcon?: boolean;
    hasShadow?: boolean;
    className?: string;
    background?: string;
}
