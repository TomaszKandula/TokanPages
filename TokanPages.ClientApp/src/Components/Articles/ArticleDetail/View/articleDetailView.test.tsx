import "../../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import { ArticleContentDto } from "../../../../Api/Models";
import { ArticleDetailView } from "./articleDetailView";

describe("test articles group component: ArticleDetailView", () => {
    it("should render correctly '<ArticleDetailView />' when content is loaded.", () => {
        const testContent: ArticleContentDto = {
            content: {
                language: "eng",
                button: "Read now",
                textReadCount: "Read count:",
                textFirstName: "First name:",
                textSurname: "Last name:",
                textRegistered: "Registered at:",
                textLanguage: "Article language:",
                textReadTime: "Read time:",
                textPublished: "Published at:",
                textUpdated: "Updated at:",
                textWritten: "Written by",
                textAbout: "About the author:"
            }
        };

        const tree = shallow(
            <ArticleDetailView
                backButtonHandler={jest.fn()}
                articleReadCount={0}
                openPopoverHandler={jest.fn()}
                closePopoverHandler={jest.fn()}
                renderSmallAvatar={<div>renderSmallAvatar</div>}
                renderLargeAvatar={<div>renderLargeAvatar</div>}
                authorAliasName="Ester"
                popoverOpen={false}
                popoverElement={null}
                authorFirstName="Ester"
                authorLastName="Exposito"
                authorRegistered=""
                articleReadTime="4"
                articleCreatedAt="2020-01-10T12:15:15"
                articleUpdatedAt="2020-01-10T12:15:15"
                articleContent={<div>articleContent</div>}
                renderLikesLeft="25"
                thumbsHandler={jest.fn()}
                totalLikes={50}
                renderAuthorName="Ester Exposito"
                authorShortBio="Happy developer"
                flagImage="eng.png"
                content={testContent}
            />
        );

        expect(tree).toMatchSnapshot();
    });
});
