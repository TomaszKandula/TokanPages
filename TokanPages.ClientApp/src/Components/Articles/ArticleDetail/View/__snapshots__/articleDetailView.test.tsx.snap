// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test articles group component: ArticleDetailView should render correctly '<ArticleDetailView />' when content is loaded. 1`] = `
<section
  className="makeStyles-section-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <Styled(MuiBox)
      py={12}
    >
      <div
        data-aos="fade-down"
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={6}
          >
            <WithStyles(ForwardRef(IconButton))
              className="makeStyles-icon-3"
              onClick={[MockFunction]}
            >
              <Memo(ForwardRef(ArrowBackIcon)) />
            </WithStyles(ForwardRef(IconButton))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-grid_read_count-9"
            item={true}
            xs={6}
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-text_read_count-10"
              component="p"
              variant="subtitle1"
            >
              Read count:
               
              0
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Divider))
          className="makeStyles-dividerTop-4"
        />
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Styled(MuiBox)
              onMouseEnter={[MockFunction]}
              onMouseLeave={[MockFunction]}
            >
              <div>
                renderSmallAvatar
              </div>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={true}
            zeroMinWidth={true}
          >
            <WithStyles(ForwardRef(Typography))
              align="left"
              className="makeStyles-aliasName-11"
              component="div"
              variant="subtitle1"
            >
              <Styled(MuiBox)
                fontWeight="fontWeightBold"
              >
                Ester
              </Styled(MuiBox)>
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Popover))
              anchorEl={null}
              anchorOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "bottom",
                }
              }
              className="makeStyles-popover-16"
              disableRestoreFocus={true}
              id="mouse-over-popover"
              onClose={[MockFunction]}
              open={false}
              transformOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
            >
              <Styled(MuiBox)
                mb={2}
                ml={3}
                mr={3}
                mt={2}
              >
                <WithStyles(ForwardRef(Typography))
                  color="textSecondary"
                  component="p"
                  variant="subtitle2"
                >
                  First name:
                   
                  Ester
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  color="textSecondary"
                  component="p"
                  variant="subtitle2"
                >
                  Last name:
                   
                  Exposito
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  color="textSecondary"
                  component="p"
                  variant="subtitle2"
                >
                  Registered at:
                   
                </WithStyles(ForwardRef(Typography))>
              </Styled(MuiBox)>
            </WithStyles(ForwardRef(Popover))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <Styled(MuiBox)
          mb={5}
          mt={1}
        >
          <div
            className="makeStyles-text_block-6"
          >
            <WithStyles(ForwardRef(Typography))
              component="p"
              variant="subtitle1"
            >
              Article language:
               
            </WithStyles(ForwardRef(Typography))>
            <img
              alt=""
              className="makeStyles-flag_image-8"
              src="http://localhost:5000/api/v1/content/assets/getNonVideoAsset?blobName=images/flags/eng.png"
            />
          </div>
          <WithStyles(ForwardRef(Typography))
            component="p"
            variant="subtitle1"
          >
            Read time:
          </WithStyles(ForwardRef(Typography))>
          <div
            className="makeStyles-text_block-6"
          >
            <WithStyles(ForwardRef(Typography))
              component="p"
              variant="subtitle1"
            >
              Published at:
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-text_padding_left-7"
              component="p"
              variant="subtitle1"
            >
              01/10/2020, 12:15 PM
            </WithStyles(ForwardRef(Typography))>
          </div>
          <div
            className="makeStyles-text_block-6"
          >
            <WithStyles(ForwardRef(Typography))
              component="p"
              variant="subtitle1"
            >
              Updated at:
            </WithStyles(ForwardRef(Typography))>
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-text_padding_left-7"
              component="p"
              variant="subtitle1"
            >
              01/10/2020, 12:15 PM
            </WithStyles(ForwardRef(Typography))>
          </div>
        </Styled(MuiBox)>
      </div>
      <div
        data-aos="fade-up"
      >
        <div>
          articleContent
        </div>
      </div>
      <Styled(MuiBox)
        mt={5}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Tooltip))
              arrow={true}
              title={
                <span
                  className="makeStyles-likesTip-15"
                >
                  <Emoji
                    text="25"
                  />
                </span>
              }
            >
              <Memo(ForwardRef(ThumbUpIcon))
                className="makeStyles-thumbsMedium-14"
                onClick={[MockFunction]}
              />
            </WithStyles(ForwardRef(Tooltip))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={true}
            zeroMinWidth={true}
          >
            <WithStyles(ForwardRef(Typography))
              component="p"
              variant="subtitle1"
            >
              50
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </Styled(MuiBox)>
      <WithStyles(ForwardRef(Divider))
        className="makeStyles-dividerBottom-5"
      />
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <div>
            renderLargeAvatar
          </div>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
          zeroMinWidth={true}
        >
          <WithStyles(ForwardRef(Typography))
            align="left"
            className="makeStyles-aliasName-11"
            color="textSecondary"
            component="span"
            variant="h6"
          >
            Written by
          </WithStyles(ForwardRef(Typography))>
          <Styled(MuiBox)
            fontWeight="fontWeightBold"
          >
            <WithStyles(ForwardRef(Typography))
              align="left"
              className="makeStyles-aliasName-11"
              component="span"
              variant="h6"
            >
              Ester Exposito
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
          <WithStyles(ForwardRef(Typography))
            align="left"
            className="makeStyles-aliasName-11"
            color="textSecondary"
            component="span"
            variant="subtitle1"
          >
            About the author:
             
            Happy developer
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
